CountlySDKLimitsConfig.m

```diff
@@ -0,0 +1,126 @@
+//  CountlySDKLimitsConfig.m
+//
+// This code is provided under the MIT License.
+//
+// Please visit www.count.ly for more information.
+
+#import "CountlySDKLimitsConfig.h"
+#import "CountlyCommon.h"
+
+
+const NSInteger kCountlyMaxKeyLength = 128;
+const NSInteger kCountlyMaxValueSize = 256;
+const NSInteger kCountlyMaxSegmentationValues = 100;
+const NSInteger kCountlyMaxBreadcrumbCount = 100;
+const NSInteger kCountlyMaxStackTraceLinesPerThread = 30;
+const NSInteger kCountlyMaxStackTraceLineLength = 200;
+
+@interface CountlySDKLimitsConfig ()
+{
+    int _maxKeyLength;
+    int _maxValueSize;
+    int _maxSegmentationValues;
+    int _maxBreadcrumbCount;
+    int _maxStackTraceLinesPerThread;
+    int _maxStackTraceLineLength;
+}
+@end
+@implementation CountlySDKLimitsConfig
+
+
+
+- (instancetype)init
+{
+    if (self = [super init])
+    {
+        _maxKeyLength = kCountlyMaxKeyLength;
+        _maxValueSize = kCountlyMaxValueSize;
+        _maxSegmentationValues = kCountlyMaxSegmentationValues;
+        _maxBreadcrumbCount = kCountlyMaxBreadcrumbCount;
+        _maxStackTraceLinesPerThread = kCountlyMaxStackTraceLinesPerThread;
+        _maxStackTraceLineLength = kCountlyMaxStackTraceLineLength;
+    }
+    
+    return self;
+}
+
+- (void)setMaxKeyLength:(int)maxKeyLength
+{
+    if (maxKeyLength < 1) {
```

Comment:
just a question? why not we setting as default if negative number is given? Is not setting it as 1 breaks all things?

Reply:
Good catch 👍 
@turtledreams What do you think about it? :)

Reply:
yeap true seems like I put out some template code here
original check should accept the value if only its greater than 0